src/libcore/any.rs

```diff
@@ -450,3 +450,26 @@ impl TypeId {
         }
     }
 }
+
+/// Returns the name of a type as a string.
+///
+/// # Note
+///
+/// This is intended for diagnostic use. The exact contents and format of the
+/// string are not specified, other than being a best-effort description of the
+/// type. For example, `type_name::<Option<String>>()` could return the
+/// `"Option<String>"` or `"std::option::Option<std::string::String>"`, but not
+/// `"foobar"`. In addition, the output may change between versions of the
+/// compiler.
```

Comment:
One thing we could additionally note here is that `type_name` may return the same result for different inputs. The easiest example is `type_name::<&'static str>()` vs `type_name::<&'a str>()` which both just return `&str`, but other examples are likely possible (e.g. multiple crates with the same name having a type of the same name but still being inherently different).

Reply:
Done!